numba/targets/arraymath.py

```diff
@@ -2867,6 +2867,17 @@ def np_imag_impl(a):
 #----------------------------------------------------------------------------
 # Misc functions
 
+@overload(np.alen)
+def np_alen(a):
+    if not type_can_asarray(a):
+        raise errors.TypingError("The argument to np.shape must be array-like")
```

Comment:
The function being typed is `np.alen` not `np.shape`?

Reply:
Also, `np.alen` seems to accept a very broad range of things...
```
In [12]: np.alen('aaa')                                                                                                         
Out[12]: 3

In [13]: np.alen(np)                                                                                                            
Out[13]: 1

In [14]: np.alen(np.alen)                                                                                                       
Out[14]: 1
```
some of which are very strange, I don't think the bizarre things are worth considering, however, strings and perhaps other types in numba's type system that can be accepted should be.

Reply:
I've implemented `alen` for strings.